### PR TITLE
Packages

### LINKED ISSUE #86: Use packages
Currently we store every file into  `server/` in package `main`. I would like to split these files into multiple packages:
- `main`: Just contains `main.go`
- `plugin`: Contains every hock
- `utils`: Self explaining
- `model?`:  contains `poll.go`. I'm not sure with the name. Its a fine name but duplicate  with `mattermost/mattermost-server/model`. But I don't want to name it `poll` to leave room for more.

### PR DESCRIPTION
**Pull request by hanzei**
This PR introduces some go packages:
- plugin
- utils
- poll

It also includes a ton of refactoring and small changes. It can be bested reviewed via the individual commits.

Fixes #86

### CODE REVIEW

**Comment by hanzei** [comment]
@kaakaa This will be my last refactoring PR. I know they are a pain to review and create chaos. After this is done, I will stop. I promise :wink:

File: server/poll/poll.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+// Encode returns a poll as a byte array
+func (p *Poll) Encode() []byte {
+	b, _ := json.Marshal(p)
+	return b
+}
+
+// Decode tries to create a poll from a byte array
+func Decode(b []byte) *Poll {
```
[the remark this answers is not in the document]
Comment (hanzei) [reply server/poll/poll.go]: Good suggestion, done.

File: server/plugin/api_test.go [hunk 2]
```diff
@@ -94,18 +98,23 @@ func TestServeFile(t *testing.T) {
...
-	poll1_out.UpdateVote("userID1", 0)
+	pollID := testutils.GetPollID()
+
+	poll1In := testutils.GetPoll()
+	poll1Out := poll1In.Copy()
+	err := poll1Out.UpdateVote("userID1", 0)
+	require.Nil(t, err)
 	expectedPost1 := &model.Post{}
-	model.ParseSlackAttachment(expectedPost1, poll1_out.ToPostActions(samplesiteURL, samplePollID, "John Doe"))
+	model.ParseSlackAttachment(expectedPost1, poll1Out.ToPostActions(testutils.GetSiteURL(), PluginId, pollID, "John Doe"))
```
[the remark this answers is not in the document]
Comment (hanzei) [reply server/plugin/api_test.go]: No, there is not particular reason. I fixed this in all files.

**Comment by hanzei** [comment]
Thanks for the feedback. I addressed your comments.

**Comment by hanzei** [comment]
Last question before merging: What are you thoughts on the renaming of `PollSettings`?

**Comment by kaakaa** [comment]
Ah...sorry, I missed it.
I think that it doesn’t need renaming. Because `PollSettings` is more specific and is not too long.

**Pull request merged**